design/queue.md

```diff
@@ -0,0 +1,26 @@
+# Executor Queue
+
+## Context
+
+This is an executor plugin for Screwdriver that makes use of [Resque][node-resque-URL] to add a queueing mechanism.
+
+## Architecture Overview
+
+![](./diagrams/executor-queue-architecture.puml.png)
+
+## Design Decisions
+
+### Router
+
+The `executor-queue` will sit behind `executor-router` until it works as we expect, so as not to disturb people already using `executor-router`.
```

Comment:
Hm I figured this could be more of a live document, so whatever is present on the doc now is what is actually in productions, and when changes are made, the document is updated along with the code.

Though I suppose whatever @petey responds to Darren's comment above would answer your question too.